docs/API.md

```diff
@@ -0,0 +1,285 @@
+* 409 Conflict  
+For any failed validations. The response body is the action entity created, with the failed validations. The DAG will not begin execution in this case.  
+
+
+### /v1.0/actions/{action_id} 
+Each action will be assigned an unique id that can be used to get details for the action, including the execution status.
+
+#### Payload Structure
+All actions will include fields that indicate the following data:
+* id (assigned during POST)
+* name - the name of the action - likely to be the DAG Names, but may be mapped for "nicer" values.
+* parameters - a dictionary of the parameters configuring the action.
+* tracking info - (user, time, etc)
+* action_lifecycle value:
+  * Pending
+  * Validation Failed
+  * Processing 
+  * Complete
+  * Failed
+* DAG_status - representing the status that airflow provides for an executing DAG
+* validations - a list of validations that have been done, including any status information for those validations. During the lifecycle of the DAG, this list of validations may continue to grow.
+* steps - the list of steps for this action, including the status for that step.
+
+```
+{ TBD }
+```
+
+#### GET
+Returns the action entity for the specified id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/validationDetails/{validation_id} 
+Allows for drilldown to validation detailed info.
+
+#### Payload Structure
+The detailed information for a validation
+```
+{ TBD }
+```
+
+#### GET
+Returns the validation detail by Id for the supplied action Id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/steps/{step_id}
+Allow for drilldown to step information. The step information includes details of the steps excution, successful or not, and enough to facilitate troubleshooting in as easy a fashion as possible. 
+
+#### Payload Structure
+The detailed information representing a single step of execution as part of an action
+```
+{ TBD }
+```
+
+#### GET
+Returns the details for a step by id for the given action by Id.
+##### Responses
+* 200 OK 
+
+
+### /v1.0/actions/{action_id}/{control_verb}
+Allows for issuing DAG controls against an action.
+
+#### Payload Structure
+None, there is no associated body for this resource
+
+#### POST 
+Trigger a control action against an activity.- this includes: pause, unpause
+##### Responses
+* 202 Accepted
+
+---
+## <a name="AirflowMonitoringAPI"></a> Airflow Monitoring API
+Airflow has a primary function of scheduling DAGs, as opposed to Shipyard's primary case of triggering DAGs.  Shipyard will need to provide functionality to allow for an operator to monitor and review the scheduled tasks.  This API will allow for accessing Airflow DAGs of either type.
+
+### /v1.0/completedDAGs
+The resource that represents DAGs in airflow
+
+#### Payload Structure
+A list of objects representing the DAGs that have run in airflow.
+```
+[ 
+  {TBD},
+  ...
+]
+```
+
+#### GET 
+Queries airflow for recent completed DAGs (a summary).
```

Comment:
I'll open this up. There is no reason that this should be restricted. This should be a "peek" into airflow, without specific regard to what went on through shipyard.